include/drivers/flash.h

```diff
@@ -121,6 +120,8 @@ static inline int z_impl_flash_read(struct device *dev, off_t offset, void *data
 /**
  *  @brief  Write buffer into flash memory.
  *
+ *  All flash drivers support source buffer located either in RAM or SoC flash.
```

Comment:
For complete clarity this should include the same text on alignment and sizes that was added for read.

Reply:
check one more, propose redaction as I'm not english native

Reply:
How about for read:

> All flash drivers support reads without alignment restrictions on the read offset, the read size, or the destination address.

and for write:

> All flash drivers support a source buffer located either in RAM or SoC flash, without alignment restrictions on the source address, or write size or offset.

?